javascript/src/seedsource/containers/Upload.js

```diff
@@ -0,0 +1,152 @@
+            files = event.dataTransfer.files
+        } else if (event.target) {
+            files = event.target.files
+        }
+
+        let zipFile
+        let shpFile
+        let prjFile
+        for (let i = 0; i < files.length; i += 1) {
+            const file = files[i]
+            if (file.name.match(/\.zip/i)) {
+                zipFile = file
+                break
+            } else if (file.name.match(/\.shp$/i)) {
+                shpFile = file
+            } else if (file.name.match(/\.prj$/i)) {
+                prjFile = file
+            }
+        }
+
+        if (zipFile) {
+            const reader = new FileReader()
+            reader.onerror = (e => {
+                this.setState({isLoading: false})
+                this.props.sendError('Error', 'Could not read the zip file.', e.message)
+                return
+            })
+            reader.onload = (e => {
+                shp(e.target.result)
+                    .then((geojson) => {
+                        this.setState({isLoading: false})
+                        this.props.onFileUpload(this.props.index, geojson, zipFile.filename)
+                        return
+                    })
+                    .catch((error) => {
+                        this.setState({isLoading: false})
+                        this.props.sendError('Error', 'Could not read the zip file.', error.message)
+                        return
+                    })
+            })
+            reader.readAsArrayBuffer(zipFile)
+        } else if (shpFile) {
+            const shpPromise = new Promise((resolve, reject) => {
+                const reader = new FileReader()
+                reader.onerror = (e) => {
+                    reject(e)
+                }
+                reader.onload = (e) => {
+                    resolve({
+                        shp: e.target.result,
+                        name: shpFile.name
+                    })
+                }
+                reader.readAsArrayBuffer(shpFile)
+            })
+            const prjPromise = new Promise((resolve, reject) => {
+                if (prjFile) {
+                    const reader = new FileReader()
+                    reader.onerror = (e) => {
+                        reject(e)
+                    }
+                    reader.onload = (e) => {
+                        resolve({
+                            prj: e.target.result
+                        })
+                    }
+                    reader.readAsArrayBuffer(prjFile)
+                } else {
+                    resolve({
+                        prj: null,
+                        warning: 'No projection file found. Assuming WGS84. If you see the wrong projection, try including the prj file.'
+                    })
+                }
+            })
+            Promise.all([shpPromise, prjPromise])
+                .then(results => {
+                    let parsedShp = shp.parseShp(results[0].shp, results[1].prj)
```

Comment:
Consider assigning `shpResult` and `prjResult` and then using them by name instead of `results[0]` and `results[1]`.

```javascript
let [shpResult, prjResult] = results
```